Tribler/community/gigachannel/request.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import absolute_import
+
+from binascii import hexlify
+
+from Tribler.pyipv8.ipv8.requestcache import RandomNumberCache
+
+
+class SearchRequestCache(RandomNumberCache):
+    """
+    This request cache keeps track of all outstanding search requests within the GigaChannelCommunity.
+    """
+    def __init__(self, request_cache, peers):
+        super(SearchRequestCache, self).__init__(request_cache, u"remote-search-request")
+        self.request_cache = request_cache
+        self.requested_peers = {hexlify(peer.mid): False for peer in peers}
+
+    @property
+    def timeout_delay(self):
+        return 10.0
+
+    def on_timeout(self):
+        self.remove_request()
+
+    def process_peer_response(self, peer):
+        """
+        Checks if a peer is expected and whether it is already processed. Further, it also checks if request is
+        complete, in which case the request is removed from the cache.
+        """
+        mid = hexlify(peer.mid)
+        if mid in self.requested_peers and not self.requested_peers[mid]:
+            self.requested_peers[mid] = True
+
+            # Check if all expected responses are received
+            if all(self.requested_peers.values()):
+                self.remove_request()
+
+            return True
```

Comment:
Personally, I don't like these side effects of a method that seems to only process stuff to me. The exact meaning of these boolean return values are not immediately clear for developers. Can we avoid returning anything here?

Reply:
This result of this method is used to decide if the response is to be processed in the community. I can break it two methods if you prefer.